fix(linter): parse regex inside `new RegExp()` with parentheses

Sysix opened the pull request:
found here:

https://github.com/eslint/eslint/blob/1b3881d7e859bec9589e39888656c33c914a8302/tests/lib/rules/no-misleading-character-class.js#L1454-L1507

Sysix commented:
* **#13448** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13448?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13448?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13448).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/utils/regex.rs:
```diff
@@ -20,45 +20,36 @@ where
+where
+    M: FnOnce(&Pattern<'_>, Span),
+{
+    let arg1 = arg1.map(|arg| arg.to_expression().get_inner_expression());
+    let arg2 = arg2.map(|arg| arg.to_expression().get_inner_expression());
```
Only when the map function is getting bigger. Not everything needs to have the DRY principles

camc314 reviewed (approved):
💪

camc314 commented:
### Merge activity

* **Sep 2, 2:37 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Sep 2, 2:37 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
fix(linter): parse regex inside `new RegExp()` with parentheses (#13448)

found here:

https://github.com/eslint/eslint/blob/1b3881d7e859bec9589e39888656c33c914a8302/tests/lib/rules/no-misleading-character-class.js#L1454-L1507
